hypervisor/src/arch/x86/mod.rs

```diff
@@ -306,6 +309,24 @@ pub struct MsrEntry {
     pub data: u64,
 }
 
+/// Error that may be returned when attempting to enable AMX state components for guests
+#[derive(Debug, Error)]
+pub enum AmxGuestSupportError {
+    /// Attempted to enable AMX on a CPU vendor that does not support this feature.
+    #[error("The host CPU's vendor does not support AMX features. Only Intel provides such CPUs.")]
+    VendorDoesNotSupportAmx,
+    /// Unable to verify that the host supports AMX.
+    #[error("The host does not support AMX tile state components: errno {errno}")]
+    AmxNotSupported { errno: NonZeroI64 },
```

Comment:
`NonZeroI64` might be a little over-engineered here but is fine with me :)

Reply:
Yeah I will just use an `i64` instead.